internal/controller/backend_security_policy.go

```diff
@@ -207,6 +225,10 @@ func getBackendSecurityPolicyAuthOIDC(spec aigv1a1.BackendSecurityPolicySpec) *e
 			return &spec.AzureCredentials.OIDCExchangeToken.OIDC
 		}
 		return nil
+	case aigv1a1.BackendSecurityPolicyTypeGCPCredentials:
```

Comment:
what if it is nil? we dont return or log anything - is that what we want?

Reply:
```
if spec.GCPCredentials != nil {
	return &spec.GCPCredentials.WorkLoadIdentityFederationConfig.WorkloadIdentityProvider.OIDCProvider.OIDC
}
```

you mean what if gcp credential is nil?
This function would just return nil
the caller will be expected to handle nil since this func returns a pointer